community/io/src/test/java/org/neo4j/graphdb/mockfs/EphemeralFileSystemAbstraction.java

```diff
@@ -1131,33 +1089,24 @@ private void copyByteBufferContents( ByteBuffer from, ByteBuffer to )
          */
         private ByteBuffer allocate( int sizeIndex )
         {
-            for ( int enlargement = 0; enlargement < 2; enlargement++ )
+            int capacity = (sizeIndex < SIZES.length) ?
+                           SIZES[sizeIndex] : ((sizeIndex - SIZES.length + 1) * SIZES[SIZES.length - 1]);
+            try
             {
-                AtomicReferenceArray<Queue<Reference<ByteBuffer>>> pools = POOLS;
-                if ( sizeIndex + enlargement < pools.length() )
+                return ByteBuffer.allocateDirect( capacity );
+            }
+            catch ( OutOfMemoryError oom )
+            {
+                try
                 {
-                    Queue<Reference<ByteBuffer>> queue = pools.get( sizeIndex + enlargement );
-                    if ( queue != null )
-                    {
-                        for (; ; )
-                        {
-                            Reference<ByteBuffer> ref = queue.poll();
-                            if ( ref == null )
-                            {
-                                break;
-                            }
-                            ByteBuffer buffer = ref.get();
-                            if ( buffer != null )
-                            {
-                                return buffer;
-                            }
-                        }
-                    }
+                    return ByteBuffer.allocate( capacity );
+                }
+                catch ( OutOfMemoryError secondOom )
+                {
+                    secondOom.addSuppressed( oom );
+                    throw secondOom;
```

Comment:
This is backwards, typically you would throw the original exception, and have the outer exception suppress the inner exception. Here you suppress the outer exception with the inner one, which doesn't seem natural.